scripts/generate-cluster-creds.sh

```diff
@@ -89,23 +89,14 @@ EOF
     rm kubeconfig.yaml
   fi
 
-  yaml_creds=$(yq '.k8s_config' /tmp/temp_k8sconfig.yaml)
-  json_creds=$(yq e -o json '.k8s_config' /tmp/temp_k8sconfig.yaml | jq -r -M -c .)
-
+  output=$(yq '.k8s_config' /tmp/temp_k8sconfig.yaml)
   # NOTE: Write to ci/ files as these files will be used in ci tests! Consider looping through all files
   # in ci dir
   for APP in merlin turing
   do
-    # TODO: REMOVE ALL CONDITIONAL STATEMENTS AND USE ONLY YAML CREDS ONCE TURING HAS BEEN REFACTORED
-    # Write to Merlin/ Turing image builder and env configs app chart values
-    yaml_creds="$yaml_creds" yq ".environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\") | .imageBuilder.k8sConfig |= env(yaml_creds)" -i "${SCRIPT_DIR}/../charts/${APP}/ci/ci-values.yaml"
-
-    # Write to Merlin/ Turing image builder app in the overarching CaraML chart
-    if [ $APP == "merlin" ]; then
-      yaml_creds="$yaml_creds" yq ".${APP}.environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\") | .${APP}.imageBuilder.k8sConfig |= env(yaml_creds)" -i "${SCRIPT_DIR}/../charts/caraml/ci/ci-values.yaml"
-    else
-      json_creds="$json_creds" yq ".${APP}.environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\") | .${APP}.imageBuilder.k8sConfig |= strenv(json_creds)" -i "${SCRIPT_DIR}/../charts/caraml/ci/ci-values.yaml"
-    fi
+    # Write to app chart and caraml chart ci-values.yaml
+    output="$output" yq ".environmentConfigs[0] *= load(\"/tmp/temp_k8sconfig.yaml\") | .imageBuilder.k8sConfig |= env(output)" -i "${SCRIPT_DIR}/../charts/${APP}/ci/ci-values.yaml"
```

Comment:
I think I know what's happening. We're modifying values in ci-values file for Merlin, turing, and caraml no matter where the chart change is.  This is why chart-testing is triggering the test for all three charts whenever there is a change in at least one of them. 

Can you modify this script to change the values file for the charts that have changed only?

we call this script [here](https://github.com/caraml-dev/helm-charts/blob/main/.github/workflows/test.yaml#L81) if there are changes in any of caraml, merlin, and turing charts.

Reply:
@deadlycoconuts : a tip for running these workflows in local. you can use https://github.com/nektos/act
`act -j <job_name> --debug`

Reply:
Ahh cool, that's helpful 😅 Thanks for sharing that! :D

Reply:
Thanks for the awesome find! I've just modified both the CI job and the script to only inject those cluster creds only for the changed charts. Feel free to review the modified script, thanks!

Reply:
The new script is better now. Thank you 👍